common/dynamicconfig/constants.go

```diff
@@ -1254,6 +1254,36 @@ these log lines can be noisy, we want to be able to turn on and sample selective
 		100,
 		`MatchingMaxTaskQueuesInDeployment represents the maximum number of task-queues that can be registered in a single worker deployment version`,
 	)
+	MatchingPollerScalingMinimumBacklog = NewTaskQueueIntSetting(
+		"matching.pollerScalingMinimumBacklog",
+		1,
+		`MatchingPollerScalingMinimumBacklog is the minimum backlog size that must be accumulated before 
```

Comment:
I wonder if this should be based on backlog age and not count. count is more relative because a count of 100 tasks can be anything between negligible or very large depending on the scale in which user is operating. Also given that we start with root partition stats only, this becomes a single partition count which is also dependent on the number of partitions. Number of partitions can change any time (manually now, and automatically in the future).

Reply:
Looking at `pollerScalingSyncMatchWaitTime` it gives another reason to make this threshold also a time threshold, on poller wait time (for scale down), another task wait time (for scale up).

Reply:
Well the downside with this is that you (very likely) start scaling up slower. If we make the value low I can see how that's a potentially good option though. 

I can certainly test it out and see how it behaves. It might help with avoiding overshooting too. Curious what others think.

Reply:
I like the symmetry of using time for scaling in both directions. The backlog age is basically "how long a task has waited", similar to "how long a poll has waited". Though measured slightly differently.